src/registry.rs

```diff
@@ -1203,468 +887,6 @@ impl CheckCode {
         }
     }
 
-    /// A placeholder representation of the `CheckKind` for the check.
-    pub fn kind(&self) -> CheckKind {
```

Comment:
I'm having trouble commenting in the right spots due to the size of the PR, but do you think `lint_source` and `category` should eventually be part of the `Violation` trait? It might be hard to move `lint_source` right now because we need a way to go from "enabled codes" to "sources of lint to invoke", and going from "enabled code" to `Violation` is an awkward operation (since it requires populating fields with placeholders -- although maybe this is doable now via associated functions?).

Reply:
Yes I think it probably makes sense to add `lint_source` to the `Violation` trait ... since the `lint_source` function has so few match arms, doing that manually in the future shouldn't pose any issues. And yes since the Violation trait isn't object-safe anyway we can easily add functions that don't take `&self` and therefore can be called without instantiation.

I don't like the name "category", [see my comment here](https://github.com/charliermarsh/ruff/issues/1546#issuecomment-1374397617). I'd rather call the function `origin`. In any case that function can be completely autogenerated just based on the code prefixes, so I think it makes more sense to generate it via a proc macro instead of adding it to the trait.

Reply:
Origin is fine.